contracts/modules/token/TokenController.sol

```diff
@@ -277,46 +283,65 @@ contract TokenController is ITokenController, LockHandler, MasterAwareV2 {
   }
 
+  /// @notice Retrieves the pending rewards for a given member.
+  /// @param member  The address of the member whose pending rewards are to be retrieved.
+  /// @return        The total amount of pending rewards for the given member.
   function getPendingRewards(address member) public view returns (uint) {
     (uint totalPendingAmountInNXM,,) = assessment().getRewards(member);
     uint governanceRewards = governance().getPendingReward(member);
     return totalPendingAmountInNXM + governanceRewards;
   }
 
-  /// Function used to claim all pending rewards in one tx. It can be used to selectively withdraw
-  /// rewards.
-  ///
-  /// @param forUser           The address for whom the governance and/or assessment rewards are
-  ///                          withdrawn.
-  /// @param fromGovernance    When true, governance rewards are withdrawn.
-  /// @param fromAssessment    When true, assessment rewards are withdrawn.
-  /// @param batchSize         The maximum number of iterations to avoid unbounded loops when
-  ///                          withdrawing governance and/or assessment rewards.
-  function withdrawPendingRewards(
-    address forUser,
-    bool fromGovernance,
-    bool fromAssessment,
-    uint batchSize
+  /// @notice Withdraws NXM from the Nexus platform based on specified options.
+  /// @dev    Ensure the NXM is available and not locked before withdrawal. Only set flags in `WithdrawNxmOptions` for
+  ///         withdrawable NXM. Reverts if some of the NXM being withdrawn is locked or unavailable.
+  /// @param stakingPoolDeposits        Details for withdrawing staking pools stake and rewards. Empty array to skip
+  /// @param stakingPoolManagerRewards  Details for withdrawing staking pools manager rewards. Empty array to skip
+  /// @param govRewardsBatchSize        The maximum number of iterations to avoid unbounded loops when withdrawing
+  ///                                   governance rewards.
+  /// @param withdrawAssessment         Options specifying assesment withdrawals, set flags to true to include
+  ///                                   specific assesment stake or rewards withdrawal.
+  function withdrawNXM(
+    WithdrawAssessment calldata withdrawAssessment,
+    StakingPoolDeposit[] calldata stakingPoolDeposits,
+    StakingPoolManagerReward[] calldata stakingPoolManagerRewards,
+    uint govRewardsBatchSize
   ) external whenNotPaused {
 
-    if (fromAssessment) {
-      assessment().withdrawRewards(forUser, batchSize.toUint104());
+    // assessment stake
+    if (withdrawAssessment.stake) {
+      assessment().unstakeAllFor(msg.sender);
+    }
+
+    // assessment rewards
+    if (withdrawAssessment.rewards) {
+      // pass in 0 batchSize to withdraw ALL Assessment rewards
+      assessment().withdrawRewards(msg.sender, 0);
+    }
+
+    // governance rewards
+    uint governanceRewards = governance().claimReward(msg.sender, govRewardsBatchSize);
```

Comment:
This is how we fixed the batchSize issue. We pass in 0 batchSize for `assessment.withdrawRewards`, which effectively withdraws ALL rewards.

The batchSize from the parameters (`govRewardsBatchSize `) will now only be use in governance rewards.

Reply:
is it possible that all assessment rewards withdrawal won't fit in a block? I guess that's why the batchSize argument was implemented in the first place

Reply:
There has been only 21 assessment so far so it seemed relatively safe to hardcode 0. but now in hindsight, this is a bad design.

it better to expose the assessment batchSize and let the caller (FE) have the option to withdraw all or in batches.

i will add a new `assessmentRewardsBatchSize` param. the withdrawNxm will now look like:

```solidity
  function withdrawNXM(
    WithdrawAssessment calldata withdrawAssessment,
    StakingPoolDeposit[] calldata stakingPoolDeposits,
    StakingPoolManagerReward[] calldata stakingPoolManagerRewards,
    uint assessmentRewardsBatchSize,
    uint govRewardsBatchSize
  ) external {
```